### PR TITLE
Jonee23 feat: Implement MediaItemsController REST API endpoints

### PR DESCRIPTION
**Pull request by jmckellar252-png**
This PR implements the complete MediaItemsController to provide REST API functionality for the library management system.

Changes:
- Added GET /items endpoint to retrieve all media items from the library
- Added GET /items/{id} endpoint to retrieve a specific media item by its ID
- Added POST /items endpoint to create new media items with proper validation
- Added DELETE /items/{id} endpoint to remove media items from the library
- Implemented proper HTTP status code handling 
- Added error handling with structured JSON error responses
- All endpoints follow REST conventions and handle edge cases appropriately

Testing:
- All unit tests in MediaItemsControllerTest pass successfully
- Manual testing with curl commands confirms all endpoints work correctly
- Error scenarios (invalid requests, non-existent items) handled properly
- API tested with both valid and invalid data inputs

The implementation uses the existing Library and Librarian services and integrates with the current MediaItem model structure. All endpoints are fully functional and ready for front-end integration.

### CODE REVIEW

File: lesson_19/template/style.css
Comment (anthonydmays) [inline lesson_19/template/style.css:1]: Must not have `lesson_19` work here.

File: lesson_23/api/java/api_app/src/main/java/com/codedifferently/lesson23/web/MediaItemsController.java [hunk 1]
```diff
@@ -31,4 +35,50 @@ public ResponseEntity<GetMediaItemsResponse> getItems() {
...
+      }
+
+      MediaItem item = MediaItemRequest.asMediaItem(request.getItem());
+      library.addMediaItem(item, librarian);
+      
+      MediaItemResponse responseItem = MediaItemResponse.from(item);
+      var response = CreateMediaItemResponse.builder().item(responseItem).build();
+      return ResponseEntity.ok(response);
+    } catch (Exception e) {
+      var errorResponse = java.util.Map.of("errors", java.util.List.of("Invalid item data"));
```
Comment (anthonydmays) [inline lesson_23/api/java/api_app/src/main/java/com/codedifferently/lesson23/web/MediaItemsController.java:67]: Look into `@ValidRequest` and you'll see that you don't need this particular catch block code at all.

File: lesson_23/api/java/api_app/src/main/java/com/codedifferently/lesson23/web/MediaItemsController.java [hunk 2]
```diff
@@ -31,4 +35,50 @@ public ResponseEntity<GetMediaItemsResponse> getItems() {
...
+    MediaItem item = items.iterator().next();
+    MediaItemResponse responseItem = MediaItemResponse.from(item);
+    return ResponseEntity.ok(responseItem);
+  }
+
+    @PostMapping("/items")
+  public ResponseEntity<?> addItem(@RequestBody CreateMediaItemRequest request) {
+    try {
+      if (request.getItem() == null) {
+        var errorResponse = java.util.Map.of("errors", java.util.List.of("Item is required"));
```
Comment (anthonydmays) [inline lesson_23/api/java/api_app/src/main/java/com/codedifferently/lesson23/web/MediaItemsController.java:56]: `@ValidRequest` will validate this for you.
